tests/5.0/requires/test_requires_unified_shared_memory_heap_map.F90

```diff
@@ -0,0 +1,86 @@
+!===--- test_requires_unified_shared_memory_heap_map.F90 --------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test checks for unified shared memory of an array that is allocated on 
+! the heap and that is accessed from host and device with the same pointer.
+!
+! The mapping of a pointer under shared memory should allow for the pointer to
+! have the same value as in the host.
+!
+!===------------------------------------------------------------------------------===//
+
+#define OMPVV_MODULE_REQUIRES_LINE !$omp requires unified_shared_memory
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_requires_unified_shared_memory_heap_map
+   USE iso_fortran_env
+   USE ompvv_lib
+   USE omp_lib
+   implicit none
+   LOGICAL:: isOffloading
+
+!$omp requires unified_shared_memory
+
+  OMPVV_TEST_AND_SET_OFFLOADING(isOffloading)
+
+  OMPVV_WARNING_IF(.NOT. isOffloading, "With no offloading, unified shared memory is guaranteed due to host execution")
+
+   OMPVV_TEST_VERBOSE(unified_shared_memory_heap_map() .NE. 0)
+
+   OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION unified_shared_memory_heap_map()
+    INTEGER:: errors, i
+    INTEGER, ALLOCATABLE:: anArray(:)
+    INTEGER, DIMENSION(N):: anArrayCopy
+
+    OMPVV_INFOMSG("Unified shared memory testing - Array on heap")
+
+    errors = 0
+
+    ALLOCATE(anArray(N))
+
+    OMPVV_ERROR_IF(.NOT. ALLOCATED(anArray), "Memory was not properly allocated")
```

Comment:
The test should exit if the allocation failed.

Reply:
Updated.